python/cudf/dataframe/index.py

```diff
@@ -530,6 +531,31 @@ def find_label_range(self, first, last):
             end += 1
         return begin, end
 
+    def hash_index(self):
+        """Hash the given index and return a new Series
+
+        Returns
+        -------
+        Series :
+            Sequence of column names. If columns is *None* (unspecified),
+            all columns in the frame are used.
+        """
+        from cudf.dataframe.series import Series
+
+        initial_hash_values = None
+        buf = Buffer(rmm.device_array(len(self), dtype=np.int32))
+        result = NumericalColumn(data=buf, dtype=buf.dtype)
+
+        _hash = cpp_hash.hash_columns([self.as_column()],
+                                      result, initial_hash_values)
+
+        sr = Series(_hash)
+
+        # hash_columns produces negative valuesg
+        # probably can switch to np.uint32
+        # when supported by libcud
+        return abs(sr)
```

Comment:
Just to confirm, you want to return a Series here and not a subtype of Index?

Reply:
I think so -- i need to build some things in dask to properly check now